### PR TITLE
Space quota

### PR DESCRIPTION
**Pull request by JanAckermann**
## Description
Enhancement: Allow updating space quota

We have implemented a way to update the space's quota

image

## Related Issue

- Fixes https://github.com/owncloud/web/issues/6470

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingAndUpload</strong> https://drone.owncloud.com/owncloud/web/23154/66/1
The following scenarios passed on retry:
- webUIUpload/uploadEdgecases.feature:109

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingPerm3</strong> https://drone.owncloud.com/owncloud/web/23154/65/1
The following scenarios passed on retry:
- webUISharingFolderAdvancedPermissionMultipleUsers/sharedFolderWithMultipleUsersAdvancedPermissions.feature:62

File: packages/web-app-files/src/views/spaces/Project.vue [hunk 1]
```diff
@@ -482,12 +495,19 @@ export default {
...
     }
   }
 }
 </script>
 
 <style lang="scss">
+#edit-quota-modal > .oc-modal {
+  overflow: initial;
+}
+
```
Comment (JanAckermann) [inline packages/web-app-files/src/views/spaces/Project.vue:510]: Just for now

**Comment by ownclouders** [comment]
Results for <strong>oCISTrashbinUploadMoveJourney</strong> https://drone.owncloud.com/owncloud/web/23164/69/1
The following scenarios passed on retry:
- webUITrashbinRestore/trashbinRestore.feature:46

**Comment by ownclouders** [comment]
Results for <strong>oC10Files5</strong> https://drone.owncloud.com/owncloud/web/23215/20/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10IntegrationNotifications</strong> https://drone.owncloud.com/owncloud/web/23215/70/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingPublic1</strong> https://drone.owncloud.com/owncloud/web/23218/67/1
The following scenarios passed on retry:
- webUISharingPublicBasic/publicLinkPublicActions.feature:44

**Comment by JammingBen** [comment]
Works well for me, one small detail I've found: Some options are hidden because the dropdown does not exceed the modal window.

image

**Comment by JanAckermann** [comment]
@JammingBen thx for feedback, problem with dropdown is in ods,  @pascalwengerter opened a pr

**Comment by ownclouders** [comment]
Results for <strong>oC10IntegrationApp2</strong> https://drone.owncloud.com/owncloud/web/23228/72/1
The following scenarios passed on retry:
- webUIUpload/upload.feature:111

**Comment by ownclouders** [comment]
Results for <strong>oCISTrashbinUploadMoveJourney</strong> https://drone.owncloud.com/owncloud/web/23228/69/1
The following scenarios passed on retry:
- webUITrashbinDelete/trashbinDelete.feature:143

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingPublic2</strong> https://drone.owncloud.com/owncloud/web/23271/68/1
The following scenarios passed on retry:
- webUISharingPublicDifferentRoles/shareByPublicLinkDifferentRoles.feature:57

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingPerm3</strong> https://drone.owncloud.com/owncloud/web/23271/65/1
The following scenarios passed on retry:
- webUISharingFolderAdvancedPermissionMultipleUsers/sharedFolderWithMultipleUsersAdvancedPermissions.feature:65

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingPublic1</strong> https://drone.owncloud.com/owncloud/web/23271/67/1
The following scenarios passed on retry:
- webUISharingPublicBasic/publicLinkCreate.feature:190

**Comment by ownclouders** [comment]
Results for <strong>oCISTrashbinUploadMoveJourney</strong> https://drone.owncloud.com/owncloud/web/23271/69/1
The following scenarios passed on retry:
- webUITrashbinDelete/trashbinDelete.feature:56

File: packages/web-app-files/tests/unit/components/Spaces/ReadmeContentModal.spec.js [hunk 2]
```diff
@@ -11,7 +11,7 @@ localVue.use(Vuex)
 
 afterEach(() => jest.clearAllMocks())
 
-describe('editReadmeContent', () => {
+describe('ReadeContentModal', () => {
```
Comment (kulmann) [inline packages/web-app-files/tests/unit/components/Spaces/ReadmeContentModal.spec.js:14]:
```suggestion
describe('ReadmeContentModal', () => {
```

File: changelog/unreleased/enhancement-space-quota [hunk 3]
```diff
@@ -0,0 +1,6 @@
+Enhancement: Allow updating space quota
+
+We have implemented a way to update the space's quota
```
Comment (kulmann) [inline changelog/unreleased/enhancement-space-quota:3]:
```suggestion
We have implemented a way to update the quota of a space
```

File: packages/web-app-files/src/mixins/spaces/actions/editQuota.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+          handler: this.$_editQuota_trigger,
+          isEnabled: ({ resources }) => {
+            if (resources.length !== 1) {
+              return false
+            }
+
+            return resources[0].spaceQuota
+          },
+          componentType: 'oc-button',
+          class: 'oc-files-actions-edit-quota-content-trigger'
```
Comment (kulmann) [inline packages/web-app-files/src/mixins/spaces/actions/editQuota.js:28]:
```suggestion
          class: 'oc-files-actions-edit-quota-trigger'
```

File: packages/web-app-files/src/views/spaces/Project.vue [hunk 5]
```diff
@@ -9,6 +9,7 @@
           :cancel="closeReadmeContentModal"
           :space="space"
         ></readme-content-modal>
+        <quota-modal v-if="quotaModalIsOpen" :cancel="closeQuotaModal" :space="space"></quota-modal>
```
Comment (kulmann) [inline packages/web-app-files/src/views/spaces/Project.vue:12]: You could write this as a self closing tag (same for the readme content modal), which is good practice IMO.

File: packages/web-app-files/src/components/Spaces/QuotaModal.vue [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+            <div class="oc-flex oc-flex-between">
+              <span>{{ displayValue }}</span>
+              <span v-if="displayUnit" class="oc-text-muted">{{ displayUnit }}</span>
+            </div>
+            <div v-if="error" class="oc-text-input-danger">{{ error }}</div>
+          </template>
+        </oc-select>
+        <p
+          class="oc-mt-xs oc-text-meta"
+          v-text="$gettext('Select an item or enter your own value')"
```
Comment (kulmann) [inline packages/web-app-files/src/components/Spaces/QuotaModal.vue:40]:
```suggestion
          v-text="$gettext('Select a quota option or enter your own value')"
```

File: packages/web-app-files/src/components/Spaces/QuotaModal.vue [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+        .updateDrive(this.space.id, { quota: { total: this.selectedOption.value } }, {})
+        .then(({ data }) => {
+          this.cancel()
+          this.UPDATE_RESOURCE_FIELD({
+            id: this.space.id,
+            field: 'spaceQuota',
+            value: data.quota
+          })
+          this.showMessage({
+            title: this.$gettext('Space quota was edited successfully')
```
Comment (kulmann) [inline packages/web-app-files/src/components/Spaces/QuotaModal.vue:145]:
```suggestion
            title: this.$gettext('Space quota was changed successfully')
```

**Review by kulmann — CHANGES_REQUESTED** [review]
some small nitpicks 🙈

**Review by kulmann — APPROVED** [review]
Superb 👌 🥇

**Comment by ownclouders** [comment]
Results for <strong>oC10XGAPortrait3</strong> https://drone.owncloud.com/owncloud/web/23304/46/1
The following scenarios passed on retry:
- webUISharingPublicDifferentRoles/shareByPublicLinkDifferentRoles.feature:33

**Pull request merged**